discovery-provider/src/model_validator.py

```diff
@@ -127,6 +129,15 @@ def create_field_schema(cls, schema, field, model):
         except BaseException as e:
             raise e
 
+    @classmethod
+    def get_properties_for_field(cls, model, field):
+        try:
+            field_schema = cls.models_to_schema_and_fields_dict[model]['field_schema'][field]
+            return field_schema['definitions'][model]['properties'][field]
+        except KeyError:
+            logger.warning(f"Could not find type for {field} for model {model}. Defaulting to 'None'")
+            return None
+
     @classmethod
     def _get_properties_field(cls, schema, model, field):
```

Comment:
this method is the same as `get_properties_for_field` but one passes in the schema and the other grabs the schema from `models_to_schema_and_fields_dict`. can we use one or the other here?

Reply:
i think these functions do slightly different things and have two different return values. also one is used externally and the schema should not be made a requirement for that. i think it's okay to leave as two functions? especially since it's abstracted and easy enough to change if we need?